streaming-binlog/src/test/java/org/openmrs/analytics/FhirConverterTest.java

```diff
@@ -125,4 +125,63 @@ public void shouldIgnoreEventWithUnknownTable() throws Exception {
 		Mockito.verify(openmrsUtil, Mockito.times(0)).fetchFhirResource(Mockito.anyString());
 	}
 	
+	@Test
+	public void shouldGenerateFhirResourcesForTablesThatHaveBeenMappedInConfig() {
+		Map<String, String> messageBody = DebeziumTestUtil.genExpectedBody();
+		
+		String tables[] = { "obs", "encounter", "cohort", "person", "provider", "relationship", "patient", "drug", "allergy",
+		        "order", "drug_order", "test_order", "program" };
+		
+		int i;
+		for (i = 0; i < tables.length; i++) {
+			
+			final String table = tables[i];
+			Map<String, Object> messageHeaders = DebeziumTestUtil.genExpectedHeaders(Operation.UPDATE, table);
+			
+			// send events
+			eventsProducer.sendBodyAndHeaders(messageBody, messageHeaders);
+		}
+		
+		Mockito.verify(openmrsUtil, Mockito.times(tables.length)).fetchFhirResource(Mockito.anyString());
+	}
+	
+	@Test
+	public void shouldNotFetchFhirResourcesForTablesWithNoCorrespondingFhirLinkTemplatesInConfig() {
+		Map<String, String> messageBody = DebeziumTestUtil.genExpectedBody();
+		// these do not have have linkTemplates.fhir in config
+		String tables[] = { "visittype", "patient_identifier", "person_attribute" };
+		
+		int i;
+		for (i = 0; i < tables.length; i++) {
+			
+			final String table = tables[i];
+			Map<String, Object> messageHeaders = DebeziumTestUtil.genExpectedHeaders(Operation.UPDATE, table);
+			
+			// send events
+			eventsProducer.sendBodyAndHeaders(messageBody, messageHeaders);
+		}
+		
+		Mockito.verify(openmrsUtil, Mockito.times(0)).fetchFhirResource(Mockito.anyString());
+	}
+	
+	//"visit"
```

Comment:
Is this comment line needed? If yes, I think it needs to be extended?

Reply:
Noted, thanks!